eth/eventdatahandler/task.go

```diff
@@ -1,4 +1,60 @@
 package eventdatahandler
 
-// TODO: struct with name and payload
-type Task = func() error
+import (
+	"github.com/bloxapp/ssv/logging/fields"
+	"github.com/bloxapp/ssv/protocol/v2/types"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+type Task struct {
+	EventType
+	Edh    *EventDataHandler
+	Ev     ethtypes.Log
+	Shares []*types.SSVShare
+}
+
+func NewTask(EventType EventType, Edh *EventDataHandler, Ev ethtypes.Log, Shares []*types.SSVShare) *Task {
+	return &Task{EventType, Edh, Ev, Shares}
+}
+
+func (t *Task) Execute() error {
+	switch t.EventType {
+	case ValidatorAdded:
+		validatorAddedEvent, err := t.Edh.filterer.ParseValidatorAdded(t.Ev)
```

Comment:
Aren't events parsed at the EventDataHandler anyway? it would be better to pass the parsed event instead of parsing again.

Reply:
Rewrote this part to use parsed contract events at execution.